Packs/OracleCloudInfrastructure/Integrations/OracleCloudInfrastructureEventCollector/OracleCloudInfrastructureEventCollector.py

```diff
@@ -0,0 +1,387 @@
+        Further more, since the format uses \n as part of the key,
+        passing this value as a configuration parameter may result in escaped \n characters.
+
+        This function will preform the following actions on the private key parameter:
+            - Unescape the string.
+            - Remove unnecessary spaces in the string.
+
+        Example:
+        Raw Private Key parameter: -----BEGIN PRIVATE KEY-----\\n\n THIS-IS-A\\n\n PRIVATE-KEY\\n\n -----END PRIVATE KEY-----
+        Output: -----BEGIN PRIVATE KEY-----\nTHIS-IS-A\nPRIVATE-KEY\n-----END PRIVATE KEY-----
+
+        Args:
+            private_key (str): Private Key parameter.
+
+        Returns:
+            str: Private Key parameter unescaped and spaceless.
+        """
+        private_key = stringUnEscape(private_key_parameter)
+        private_key = private_key.replace('\n\n', '\n')
+
+        if ' ' not in private_key:
+            return private_key
+
+        prefix = '-----BEGIN PRIVATE KEY-----\n'
+        postfix = '\n-----END PRIVATE KEY-----'
+
+        private_key = private_key.replace(prefix, '').replace(postfix, '')
+
+        private_key_sections = private_key.strip().split(' ')
+        striped_private_key = ''.join(private_key_sections)
+        return prefix + striped_private_key + postfix
+
+
+''' Event related functions '''
+
+
+def add_time_key_to_events(events: List[dict[str, Any]]) -> List[dict[str, Any]]:
+    """
+    Add the _time key to the events.
+    Args:
+        events (List[dict[str, Any]]): The events to add the time key to.
+    Returns:
+        List[dict[str, Any]]: The events with the _time key.
+    """
+    for event in events:
+        if event_time := event.get("eventTime"):
+            event["_time"] = event_time
+
+    return events
+
+
+def get_last_event_time(events: List, first_fetch_time: datetime) -> str:
+    """Get the latest event time from the fetched events list for next fetch cycle.
+    - Given a non empty list of events, the function will return the time of the last event (most recent event) + 1 milliseconds.
+    - If the event list is empty, the function will return the current first fetch time.
+
+    Args:
+        events (List): list of fetched events.
+        first_fetch_time (datetime): current first fetch time.
+
+    Returns:
+        str: first fetch time for next fetch cycle.
+        - If the events list is not empty, return the time of the lateset event + 1 milliseconds.
+        - If the events list is empty, return the current first fetch time.
+    """
+    # if no events were fetched, return the current first fetch time.
+    if not events:
+        return first_fetch_time.strftime(DATE_FORMAT)
+
+    # get the event time of last event in the list (will always be the most recent event)
+    last_event_time = events[-1].get('eventTime')
+    if not isinstance(last_event_time, datetime):
+        last_event_time = arg_to_datetime(arg=last_event_time, settings={'RETURN_AS_TIMEZONE_AWARE': False})
```

Comment:
Maybe we can add this setting - `{'RETURN_AS_TIMEZONE_AWARE': False}` as a constant, cause I see you are using it multiple times in the code.

Reply:
I think its more readable the way it is, where you can see what settings are passed immediately instead of referring to the constant in the beginning of the code.

Reply:
You can leave it this way, it is not critical.